TT 02 grid processor

MarcoCaponera opened the pull request:
Added:
- Full Preview System
- Full Placement System

Classes:
- GridInteractComponent: player component that lets him interact with the grid
- GridProcessorComponent: base component for adding grid functionality
- GridPreviewComponent: inherits from "GridProcessorComponent", adds preview functionality to the grid
- GridPlacementComponent: inherits from "GridProcessorComponent", adds placement functionality to the grid

MikiVernox committed:
Trap Base

KuraiKunshu committed:
created the necessary component and interface for the grid interaction system

KuraiKunshu committed:
Update Line Trace Base System

KuraiKunshu committed:
fixed volume check collision

KuraiKunshu committed:
general fix to grid get surface logic

KuraiKunshu committed:
removed redundant loop in get closer surface logic

KuraiKunshu committed:
Merge pull request #11 from aiv01/TT-02-TrapBase, Merging Base Trap for Preview Logic

Mergin Base Trap for Preview Logic

KuraiKunshu committed:
last fix before merge for grid system correction

KuraiKunshu committed:
merged grid system and grid processor

KuraiKunshu committed:
added comment, implemented base Trap Datatable for Place/Preview

KuraiKunshu committed:
fix to halfgriddata

KuraiKunshu committed:
small adjustment

KuraiKunshu committed:
base preview

KuraiKunshu committed:
added negative color and mesh scale

KuraiKunshu committed:
set rotation for preview trap

KuraiKunshu committed:
fix for nullptr in grid interact, implemented place trap (need debugging), added various comment

KuraiKunshu committed:
fix for negative material after placement

MarcoCaponera committed:
Merge branch 'development' into TT-02-GridProcessor

MarcoCaponera committed:
Merge branch 'development' into TT-02-GridProcessor

MarcoCaponera committed:
map location

changed map directory

ClaudioDallai reviewed (commented):
- **Row 22 GridInteractComponent.cpp:** Maybe typo in comment ("it's" instead of "its").
- **Row 58+ GridProcessorComponent.cpp:** Some comments are in italian.
- **ETrapType declaration, GridGeneratorVolume.h:** Enums are not deriving from uint8. Not sure if this is really a problem, but we experienced some warnings if not doing so.

alex28931 reviewed (commented):
same comments made by claudio

MarcoCaponera committed:
requested changes

Solved requested changes:
- Corrected typo in  "GridInteractComponent.cpp"
- Translated italian comments in "GridProcessorComponent.cpp"
- "ESurfaceOrientation" and "ETrapType" now derive from uint8

ClaudioDallai reviewed (approved):
Everything requested was solved!

alex28931 reviewed (approved):
LGTM
